### PR TITLE
Added Batch Requests

### PR DESCRIPTION
**Pull request by AliabbasMerchant**
@YourTechBud Done

### CODE REVIEW

**Comment by YourTechBud** [comment]
@AliabbasMerchant we cannot panic if the batch conditions don't match. Let's just return an error instead. Or what you can do is, force the batch's dbType and project internally. What do you think?

**Comment by AliabbasMerchant** [comment]
I think it would be better to send an error

File: api/mgo/delete.go [hunk 1]
```diff
@@ -11,17 +11,17 @@ import (
...
+	op, db, col string
+	find        utils.M
+	config      *config.Config
 }
 
 func initDelete(ctx context.Context, db, col, op string, config *config.Config) *Delete {
 	m := &proto.Meta{Col: col, DbType: db, Project: config.Project, Token: config.Token}
 	f := make(utils.M)
-	return &Delete{ctx, m, op, f, config}
+	return &Delete{ctx, m, op, db, col, f, config}
```
Comment (YourTechBud) [inline api/mgo/delete.go]: Why store db and collection again? Its already there in the meta

File: api/mgo/batch.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	getOperation()  string
+}
+
+func initBatch(ctx context.Context, db string, config *config.Config) *Batch {
+	m := &proto.Meta{DbType: db, Project: config.Project, Token: config.Token}
+	return &Batch{ctx, m, db, config, []*proto.AllRequest{}}
+}
+
+// Add adds a delete request to batch
+func (b *Batch) Add(request *Request) (*Batch, error) {
```
Comment (YourTechBud) [inline api/mgo/batch.go]: Let the batch.add() return an error only. It need not return itself.

**Review by YourTechBud — CHANGES_REQUESTED** [review]
Minor changes. Really loved the way the code is structured!

File: api/mgo/batch.go (hunk 2 above)
Comment (AliabbasMerchant) [reply api/mgo/batch.go]: Cool

File: api/mgo/delete.go (hunk 1 above)
Comment (AliabbasMerchant) [reply api/mgo/delete.go]: Oh. Right

**Comment by AliabbasMerchant** [comment]
@YourTechBud Done

**Pull request merged**